templates/results.html

```diff
@@ -0,0 +1,87 @@
+{% extends "base.html" %}
+
+{% block title %} Results {% endblock %}
+
+{% block content %}
+<div class="container no-results-container">
+    {# -- No Results Section -- #}
+    {% if n_rows == 0 %}
+        <div class="alert alert-warning mt-4 text-center">
+            <i class="fas fa-search-minus fa-2x mb-3 text-warning"></i>
+            <h2>No Courses Found</h2>
+            <!-- Show the subject searched for, with line breaks -->
+            <p class="text-muted mb-2">{{ subject|replace('\n', '<br>')|safe }}</p>
+            <p class="lead">No courses matched your search. Please adjust your filters and try again.</p>
+            <a href="{{ url_for('index') }}" class="btn btn-search mt-3">
+                <i class="fas fa-arrow-left me-2"></i>Back to Search
+            </a>
+        </div>
+    {% else %}
+        {# -- Results Section -- #}
+        <div class="container results-container">
+            <div class="row align-items-start mb-4 mt-4 g-3 flex-md-row flex-column">
+                <div class="col-md-9">
+                    <div class="results-header p-3 h-100">
+                        <h2 class="mb-2">Course Data</h2>
+                        <!-- Show the subject description -->
+                        <div class="results-subtext text-muted">
+                            {{ subject|replace('\n', '<br>')|safe }}
+                        </div>
+                        <!-- Summary Table -->
```

Comment:
I suggest this code instead here, makes the highlighted header actually based on the `subject`.
```suggestion
                        <h2 class="mb-2">{{ subject|replace('\n', ' ')|safe }}</h2>
```